cmd/templ/generatecmd/main.go

```diff
@@ -179,20 +218,62 @@ func runCmd(ctx context.Context, w io.Writer, args Arguments) (err error) {
 				}()
 			}
 		}
-		if err = checkTemplVersion(args.Path); err != nil {
+		if err := checkTemplVersion(args.Path); err != nil {
 			logWarning(w, "templ version check failed: %v\n", err)
-			err = nil
 		}
+
 		if firstRunComplete {
 			if changesFound > 0 {
 				bo.Reset()
 			}
 			time.Sleep(bo.NextBackOff())
 		}
+
 		firstRunComplete = true
 		start = time.Now()
 	}
-	return err
+
+	return nil
+}
+
+func generateProduction(ctx context.Context, w io.Writer, args Arguments, opts []generator.GenerateOpt, p *proxy.Handler) error {
```

Comment:
I like how this is separate. It keeps the risk of breaking stuff in dev mode relatively low.